todo-app/src/App.js

```diff
@@ -1,21 +1,174 @@
+              id= {"check" + todoListItem.toDoId}
+              type="checkbox"
+              checked = {todoListItem.done}
+              onChange={this.handleDoneTodo}/> : null
+          }
+          {todoListItem.action === "Edit" ?
+            <span>done</span> : null
+          }
+          <button
+            id= {"primaryBTN" + todoListItem.toDoId}
+            style= {{flex: "1 0 0"}}
+            onClick= {this.handlePrimaryBTN}>
+              {todoListItem.action}
+          </button>
+          <button
+            id= {"secondaryBTN" + todoListItem.toDoId}
+            style= {{flex: "1 0 0"}}
+            onClick= {this.handleSecondaryBTN}>
+              {todoListItem.action === "Edit" ? "Delete" : "Cancel"}
+          </button>
+      </li>);
     return (
-      <div className="App">
-        <div className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h2>Welcome to React</h2>
+      <div>
+        <div key style = {{display: "flex"}}>
+          {this.state.deletAllIsClicked ?
+            <span style= {{flex: "6 0 0"}}>
+              Are you sure that you want to permanently delete all the todos?
+            </span> :
+            <input
+              id= "newTodoDescription"
+              type= "text"
+              placeholder= "Please write here the description for the new todo that you want to add"
+              style= {{flex: "6 0 0"}}/>
+          }
+          <button style= {{flex: "1 0 0"}} onClick= {this.handleAddOrRemoveAllBTN}>
+            {this.state.deletAllIsClicked ? "Delete all" : "Add todo"}
+          </button>
+          {/*The phe previous button adds a todo or deletes all of them depending on the value
+          of this.state.deletAllIsClicked. This value is determined by the next button*/}
+          <button style= {{flex: "1 0 0"}} onClick= {this.handleControllerBTN}>
+            {this.state.deletAllIsClicked ? "Cancel" : "Delete all"}
+          </button>
         </div>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+        <ul style= {{padding: "0 0 0 0"}}>
+          {todosList}
+        </ul>
       </div>
-    );
+    )
   }
-}
-
-export default App;
+  handleAddOrRemoveAllBTN = () => {
+    if (this.state.deletAllIsClicked) {
+      this.setState({
+        todos: [],
+        nextId: 1,
+        deletAllIsClicked: false
+      });
+    }
+    else {
+      const newToDo = document.getElementById("newTodoDescription").value;
+      document.getElementById("newTodoDescription").value = "";
+      if (newToDo) {
+        this.setState({
+          todos: this.state.todos.concat(
+            {
+              description: newToDo,
+              done: false,
+              toDoId: this.state.nextId,
+              action: "Edit"
+            }),
+          nextId: this.state.nextId + 1,
+        });
+      }
+    }
+  }
+  handleControllerBTN = () => {
```

Comment:
I dont think that passing an argument to this function is necessary or even proper, because the only purpose of this function is to invert the state of this.setState.deletAllIsClicked (this.setState.clearIsClicked in the new version because I've changed the name).
This function decides the mission of the first button (is it gonna be add or clear?). That's why I gave it this name (ControllerBTN). But I think that (as you said) the name still doesn't clearly indicate what this function does. I'll try to find a better one